nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java

```diff
@@ -379,14 +385,19 @@ private UpdateableReference loadReference(String requestedRef, String hash) {
     try {
       Reference ref = requestedRef == null ? api.getDefaultBranch()
           : api.getReference().refName(requestedRef).get();
-      if (hash != null) {
-        if (ref instanceof Branch) {
-          ref = Branch.of(ref.getName(), hash);
-        } else {
-          ref = Tag.of(ref.getName(), hash);
-        }
+      String hashOnRef;
```

Comment:
not sure these changes are even needed. wouldn't the code behave the same way without doing any changes and just passing `initialHash` to the method?